maths/greatest_common_divisor.py

```diff
@@ -5,6 +5,15 @@
 """
 
 
+
+# This method is more efficient not acquire more memory cause is no use of any stacks like in recursive as next below mentioned. 
+def gcd_by_iterative(x,y):
+    while y:x,y=y,x%y
```

Comment:
Please split this over two lines for readability.
Please add doctests to both functions.  See CONTRIBUTING.md for details.